pkg/cloud/tags_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud_test
+
+import (
+	infrav1 "github.com/aws/cluster-api-provider-cloudstack/api/v1beta1"
+	"github.com/aws/cluster-api-provider-cloudstack/pkg/cloud"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Tag Unit Tests", func() {
+	var (
+		cluster *infrav1.CloudStackCluster
+	)
+
+	BeforeEach(func() {
+		cluster = &infrav1.CloudStackCluster{
+			Spec: infrav1.CloudStackClusterSpec{
+				Zone: "Zone1", Network: "SharedGuestNet1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				UID: "0",
+			},
+		}
+	})
+
+	Context("Tag Integ Tests", func() {
+		client, connectionErr := cloud.NewClient("../../cloud-config")
+
+		const (
+			tagKey   = "test_tag"
+			tagValue = "arbitrary_value"
+		)
+
+		var (
+			networkId string
+			testTags  map[string]string
+		)
+
+		BeforeEach(func() {
+			if connectionErr != nil { // Only do these tests if an actual ACS instance is available via cloud-config.
+				Skip("Could not connect to ACS instance.")
+			}
+
+			if err := client.GetOrCreateNetwork(cluster); err != nil {
```

Comment:
I think in these circumstances we should just create one and delete it in the after block.

I won't gate the PR for it, but it's a nice to have.

Once we get our tests running proper in the pipeline, these tests won't actually be skipped, but will hit our EC2 instance.